packages/webui/src/client/ui/Status/SystemStatus.tsx

```diff
@@ -72,433 +73,434 @@ export function statusCodeToString(t: i18next.TFunction, statusCode: StatusCode)
 	}
 }
 
-export const DeviceItem = reacti18next.withTranslation()(
-	class DeviceItem extends React.Component<Translated<IDeviceItemProps>, IDeviceItemState> {
-		deviceTypeString() {
-			const t = this.props.t
-
-			switch (this.props.device.type) {
-				case PeripheralDeviceType.MOS:
-					return t('MOS Gateway')
-				case PeripheralDeviceType.PLAYOUT:
-					return t('Play-out Gateway')
-				case PeripheralDeviceType.MEDIA_MANAGER:
-					return t('Media Manager')
-				case PeripheralDeviceType.LIVE_STATUS:
-					return t('Live Status Gateway')
-				default:
-					return t('Unknown Device')
-			}
-		}
-		deviceVersions() {
-			const versions = this.props.device.versions
-			if (versions) {
-				return _.map(versions, (version, packageName) => {
-					return packageName + ': ' + version
-				}).join('\n')
-			}
+export function DeviceItem(props: IDeviceItemProps): JSX.Element {
+	const { t, i18n, ready } = reacti18next.useTranslation()
+
+	const userPermissions = React.useContext(UserPermissionsContext)
+
+	return <DeviceItemInner {...props} t={t} tReady={ready} i18n={i18n} userPermissions={userPermissions} />
+}
+
+interface DeviceItemInnerProps extends Translated<IDeviceItemProps> {
+	userPermissions: Readonly<UserPermissions>
+}
+
+class DeviceItemInner extends React.Component<DeviceItemInnerProps, IDeviceItemState> {
```

Comment:
Considering the level of restructuring here, I think this could have been made into a Functional Component, reducing any need for this two-step `DeviceItem`/`DeviceItemInner` implementation. IDeviceItemState is empty, so making this an FC at this stage would be great.

Reply:
yeah I can do that. I was trying to avoid it as it requires a bit of unpicking the component, as it is more than just a render method, unlike `CoreItem` was.